api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/v2/EligibilityClientV2Test.java

```diff
@@ -0,0 +1,99 @@
+
+    @Mock
+    private RestTemplate restTemplate;
+
+    private EligibilityClientV2 client;
+
+    @BeforeEach
+    void setup() {
+        client = new EligibilityClientV2(BASE_URI, restTemplate);
+    }
+
+    @Test
+    void shouldCheckIdentityAndEligibilitySuccessfully() {
+        Claimant claimant = aValidClaimant();
+        IdentityAndEligibilityResponse identityAndEligibilityResponse = anIdentityMatchedEligibilityConfirmedUCResponseWithAllMatches();
+        ResponseEntity<IdentityAndEligibilityResponse> response = new ResponseEntity<>(identityAndEligibilityResponse, HttpStatus.OK);
+        given(restTemplate.postForEntity(anyString(), any(), eq(IdentityAndEligibilityResponse.class)))
+                .willReturn(response);
+
+        IdentityAndEligibilityResponse actualResponse = client.checkIdentityAndEligibility(claimant);
+
+        assertThat(actualResponse).isNotNull();
+        assertThat(actualResponse).isEqualTo(identityAndEligibilityResponse);
+        verify(restTemplate).postForEntity(FULL_URI, expectedPerson(), IdentityAndEligibilityResponse.class);
+    }
+
+    @Test
+    void shouldThrowAnExceptionWhenPostCallNotOkOrNotFound() {
+        Claimant claimant = aValidClaimant();
+        IdentityAndEligibilityResponse identityAndEligibilityResponse = anIdentityMatchedEligibilityConfirmedUCResponseWithAllMatches();
+        ResponseEntity<IdentityAndEligibilityResponse> response = new ResponseEntity<>(identityAndEligibilityResponse, HttpStatus.BAD_REQUEST);
+        given(restTemplate.postForEntity(anyString(), any(), eq(IdentityAndEligibilityResponse.class)))
+                .willReturn(response);
+
+        EligibilityClientException thrown = catchThrowableOfType(() -> client.checkIdentityAndEligibility(claimant), EligibilityClientException.class);
+
+        assertThat(thrown).as("Should throw an Exception when response code is not OK").isNotNull();
+        assertThat(thrown).hasMessage("Response code from Eligibility service was not OK, received: 400");
+        verify(restTemplate).postForEntity(FULL_URI, expectedPerson(), IdentityAndEligibilityResponse.class);
+    }
+
+    @Test
+    void shouldThrowAnExceptionWhenPostCallReturnsError() {
+        Claimant claimant = aValidClaimant();
+        given(restTemplate.postForEntity(anyString(), any(), eq(IdentityAndEligibilityResponse.class)))
+                .willThrow(new RestClientException("Test exception"));
+
+        EligibilityClientException thrown = catchThrowableOfType(() -> client.checkIdentityAndEligibility(claimant), EligibilityClientException.class);
+
+        assertThat(thrown).as("Should throw an Exception when post call returns error").isNotNull();
+        assertThat(thrown).hasMessage("Exception caught trying to call eligibility service at: " + FULL_URI);
+        verify(restTemplate).postForEntity(FULL_URI, expectedPerson(), IdentityAndEligibilityResponse.class);
+    }
+
+    private PersonDTOV2 expectedPerson() {
```

Comment:
I needed to do this because the constants used for DWP API V2 and the current ones in the claimant-service differ, I'll see if I can easily merge them without too many changes.